### ISSUE TITLE
EfficietNet + Generalized Mean Pooling causes NaN loss! 

### ISSUE THREAD

**Issue by innat** [keras-team/keras#15600]
Config:

```
TF: 2.6
Environment: Colab
```

## Issue

I've plugged a GeM pooling layer into the EfficientNet model but for an unknown reason, it causes `nan` loss. But with other image-net models, it works just fine.

**Comment by sachinprasadhs** [comment]
I'm able to see loss, please refer the gist [here](https://colab.research.google.com/gist/sachinprasadhs/f08ccaef065d1df5713ad230e7c98f50/efficietnet-generalized-mean-pooling.ipynb) and confirm the same. Thanks!

**Comment by innat** [comment]
@sachinprasadhs oh, sorry. I was tweaking on the same file. I set the `self.p` parameter not trainable. After doing that, the error was gone. But I think it should be set as `True`.  Please check now again. 

```python
 def build(self, input_shape):
        self.p = self.add_weight(name="norms", 
                                 shape=(input_shape[-1],),
                                 initializer=keras.initializers.constant(self.init_norm),
                                 trainable=True) # < --- SET AS TRUE 
        super(GeneralizedMeanPooling2D, self).build(input_shape)
```

**Comment by sachinprasadhs** [comment]
Now I can see the `nan` loss when the trainable is set to `True` in `add_weight`, attaching the Gist [here](https://colab.research.google.com/gist/sachinprasadhs/a6fef9c3d589852e70fa5b126fa06997/efficietnet-generalized-mean-pooling.ipynb) for reference.Thanks!

**Comment by mattdangerw** [comment]
Given that this bug is not hit when trainable is false, we believe this might be an issue with zeros in `self.p`, which look like they would trigger a nan. You may need to try protecting against zero values in p in the custom GeM layer.

**Comment by innat** [comment]
It's happening with the efficientnet model only and not other vision models. Okay, I'll look into it again.

**Comment by Jobayer-000** [comment]
setting the init_value to 1 solve the problem for me.

**Comment by innat** [comment]
@Jobayer-000 I think for GeM, this p needs to be trainable.
